src/partitioned_data/partitionedNLPModels/partitionedNLPModel.jl

```diff
@@ -76,4 +76,70 @@ function NLPModels.grad!(
   return g
 end
 
+"""
+    B = hess_approx(nlp::PartiallySeparableNLPModel)
+
+Return the Hessian approximation of `nlp`.
+"""
+hess_approx(nlp::PartiallySeparableNLPModel) = get_pB(nlp)
+
+function NLPModels.hess(
+  nlp::PartiallySeparableNLPModel,
+  x::AbstractVector
+)
+  increment!(nlp, :neval_hess)
+  hess(nlp.part_data, x)
+end
+
+"""
+    Hv = hprod(nlp, x, v; obj_weight=1.0)
```

Comment:
should be hprod!

Reply:
This one doesn't have `hv` as a parameter.

Reply:
I removed `hprod` to keep only `hprod!` and rely on the method `hprod` from NLPModels.